tests/robots/test_robot.py

```diff
@@ -62,5 +68,93 @@ def test_basic_name_links(ur5_links):
 
 
 def test_basic_attr():
-    robot = Robot.basic('testbot', location="rfl")
-    assert robot.model.attr['location'] == "rfl"
+    robot = Robot.basic('testbot', location='rfl')
+    assert robot.model.attr['location'] == 'rfl'
+
+
+def test_name(panda_robot_instance):
+    robot = panda_robot_instance
+    assert robot.name == 'panda'
+
+
+def test_group_names(ur5_robot_instance):
+    robot = ur5_robot_instance
+    assert robot.group_names == ['manipulator', 'endeffector']
+
+
+def test_main_group_name(panda_robot_instance):
+    robot = panda_robot_instance
+    assert robot.main_group_name == 'panda_arm_hand'
+
+
+def test_root_name(ur5_robot_instance):
+    robot = ur5_robot_instance
+    assert robot.root_name == 'world'
+
+
+@pytest.mark.parametrize('group,expected_result', [(None, 'panda_rightfinger'), ('panda_arm_hand', 'panda_rightfinger'), ('panda_arm', 'panda_link8')])
+def test_get_end_effector_link_name(panda_robot_instance, group,  expected_result):
+    robot = panda_robot_instance
+    assert robot.get_end_effector_link_name(group=group) == expected_result
```

Comment:
This one can easily be rewritten with
```suggestion
def test_get_end_effector_link_name(panda_robot_instance):
    robot = panda_robot_instance
    assert robot.get_end_effector_link_name(group=None) == 'panda_rightfinger'
    assert robot.get_end_effector_link_name(group='panda_arm_hand') == 'panda_rightfinger'
    assert robot.get_end_effector_link_name(group='panda_arm') == 'panda_link8'
```

and in my opinion, this is significantly clearer to understand.